src/api/endpoints/logged_activities/approve.py

```diff
@@ -10,75 +10,53 @@
-                            ' logged_activities at a go'), 403)
-
-            if not isinstance(logged_activities_ids, list) or \
-                    not logged_activities_ids:
+            if not isinstance(logged_activities_ids, list) or not logged_activities_ids:
                 return response_builder(dict(
                     message='A List/Array with at least one logged activity'
                             ' id is needed!'), 400)
 
-            unique_activities_ids = set([])
-            for logged_activities_id in logged_activities_ids:
-                logged_activities = self.LoggedActivity.query.get(
-                    logged_activities_id
-                )
-                if logged_activities is None or logged_activities.redeemed \
-                    or logged_activities.status in ['rejected', 'in review',
-                                                    'approved']:
-                    continue
-                unique_activities_ids.add(logged_activities)
-
-            if not unique_activities_ids:
+            if len(logged_activities_ids) > 20:
                 return response_builder(dict(
-                    status='failed',
-                    message='Invalid logged activities or no pending logged'
-                            ' activities in request'),
-                    400)
-            else:
-                for unique_activities_id in list(unique_activities_ids):
-                    unique_activities_id.status = 'approved'
-                    unique_activities_id.society.total_points = \
-                        unique_activities_id
+                    message='Sorry, you can not approve more than 20'
+                            ' logged_activities at a go'), 403)
 
-                user_logged_activities = logged_activities_schema.dump(
-                    unique_activities_ids).data
+            request_approval_data = self.LoggedActivity.query.filter(
+                    self.LoggedActivity.uuid.in_(payload['loggedActivitiesIds']),
+                    self.LoggedActivity.status.in_(['pending'])).all()
 
-                # NOTE: this code works as expected, shipping it out for the
-                # MVP further optimization will be done from line 319 - 325
-                # using marshmallow
-                for user_logged_activity in user_logged_activities:
-                    user_logged_activity['society'] = {
-                        'id': user_logged_activity['societyId'],
-                        'name': user_logged_activity['society']
-                    }
-                    del user_logged_activity['society']['id']
-                    del user_logged_activity['societyId']
+            request_approval = self.LoggedActivity.query.filter(self.LoggedActivity.uuid.in_(
+                payload['loggedActivitiesIds']),
+                self.LoggedActivity.status.in_(['pending']))\
+                .update({'status': 'approved'}, synchronize_session='fetch')
```

Comment:
Curious no. 2. What does the synchronize_session kwarg do?

Reply:
`synchronize_session=fetch` performs a select query before the update to find objects that are matched by the update query. The updated attributes are expired on matched objects.
`Check this:` http://docs.sqlalchemy.org/en/latest/orm/query.html?highlight=query.update#sqlalchemy.orm.query.Query.update.params.synchronize_session

Reply:
I've changed it to `synchronize_session=False`. Sqlalchemy says it's more efficient.

Reply:
Hehehe. I was just about to share this from their docs:
`The 'fetch' strategy results in an additional SELECT statement emitted and will significantly reduce performance.`

Reply:
I also had trouble understanding what `expired` meant in the sqlalchemy context and wound up [here](https://www.michaelcho.me/article/sqlalchemy-commit-flush-expire-refresh-merge-whats-the-difference). An excellent concise read.

Reply:
I changed it already to `False` and committed the session for it to work as expected. Nice follow up still. Thanks.